frontend/src/pages/NewRunV2.tsx

```diff
@@ -279,34 +338,24 @@ function NewRunV2(props: NewRunV2Props) {
       name: runName,
       pipeline_spec: {
         // FE can only provide either pipeline_manifest or pipeline version
-        pipeline_manifest: hasVersionID(apiRun) ? undefined : templateString,
+        pipeline_manifest: hasVersionID(cloneOrigin) ? undefined : templateString,
         runtime_config: {
           // TODO(zijianjoy): determine whether to provide pipeline root.
           pipeline_root: undefined, // pipelineRoot,
           parameters: runtimeParameters,
         },
       },
       //TODO(jlyaoyuli): deprecate the resource reference and use pipeline / workflow manifest
-      resource_references: apiResourceRefFromRun ? apiResourceRefFromRun : references,
+      resource_references: existResourceRef ? existResourceRef : references,
       service_account: serviceAccount,
     };
 
-    let newRecurringRun: ApiJob = Object.assign(
-      newRun,
-      isRecurringRun
-        ? {
-            enabled: true,
-            max_concurrency: maxConcurrentRuns || '1',
-            no_catchup: !catchup,
-            trigger: trigger,
-          }
-        : {
-            enabled: false,
-            max_concurrency: undefined,
-            no_catchup: undefined,
-            trigger: undefined,
-          },
```

Comment:
Question: Why do we remove this logic?

Reply:
The declaration of `run` object and `recurring run` object is independent in v2, so we don't have to consider the one-off `run` condition here in `recurring run` definition.

Reply:
Thank you for the explanation, I think this change actually makes newRun to always include recurring run related fields. Look at the definition of `Object.assign()`: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign

Reply:
Thank you for pointing this out!